src/test/java/com/amazon/opendistroforelasticsearch/sql/doctest/core/response/SqlResponseFormat.java

```diff
@@ -84,4 +73,63 @@ public String format(SqlResponse sqlResponse) {
      */
     public abstract String format(SqlResponse sqlResponse);
 
+    /**
+     * Note that we put this format() here because it's shared by two format enums.
+     * @param sqlResponse   original response from plugin
+     * @param isSorted      true to sort the result or just leave it as is
+     */
+    protected String format(SqlResponse sqlResponse, boolean isSorted) {
+        JSONObject body = new JSONObject(sqlResponse.body());
+        if (body.isNull("schema")) {
+            throw new IllegalStateException(
+                "Only JDBC response can be formatted to table: " + sqlResponse.body());
+        }
+
+        Object[] header = parseHeader(body.getJSONArray("schema"));
+        List<Object[]> rows = parseDataRows(body.getJSONArray("datarows"), isSorted);
+
+        DataTable table = new DataTable(header);
+        for (Object[] row : rows) {
+            table.addRow(row);
+        }
+        return table.toString();
+    }
+
+    private Object[] parseHeader(JSONArray schema) {
+        Object[] header = new Object[schema.length()];
+        for (int i = 0; i < header.length; i++) {
+            JSONObject nameType = schema.getJSONObject(i);
+            header[i] = nameType.optString("alias", nameType.getString("name"));
+        }
+        return header;
+    }
+
+    private List<Object[]> parseDataRows(JSONArray rows, boolean isSorted) {
+        List<Object[]> rowsToSort = new ArrayList<>();
+        for (Object row : rows) {
+            rowsToSort.add(((JSONArray) row).toList().toArray());
+        }
+
+        if (isSorted) {
+            sort(rowsToSort);
+        }
+        return rowsToSort;
+    }
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
```

Comment:
minor, Implement Ordering seems much readable.

Reply:
Good suggestion. I tried it out but our case seems different (sorting 2 lists of objects and `Ordering` only works for object comparison). Will refactor this part if I can find how to covert it to list comparator later. Thanks!